specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                wil::unique_bstr domain = GetDomainOfUri(uri.get());
+                
+                wil::com_ptr<ICoreWebView2Settings> m_settings;
+                CHECK_FAILURE(m_webView->get_Settings(&m_settings));
+                wil::com_ptr<ICoreWebView2Settings12> coreWebView2Settings12;
+                coreWebView2Settings12 = m_settings.try_query<ICoreWebView2Settings12();
+                CHECK_FEATURE_RETURN(coreWebView2Settings12);
+                
+                BOOL enabled;
+                CHECK_FAILURE(coreWebView2Settings12->get_IsNonClientRegionSupportEnabled(&enabled));
+
+                if (_wcsicmp(domain.get(), allowedHostName) == 0 && !enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(TRUE));
+                }
+                else if (_wcsicmp(domain.get(), allowedHostName) != 0 && enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(FALSE));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_navigationStartingToken));
+}
+```
+## .NET C#
+```c#
+// WebView2 control is defined in the xaml
+// <wv2:WebView2 x:Name="webView" Source="https://www.microsoft.com/"/>
+public MainWindow() 
+{
+    InitializeComponent();
+    webView.NavigationStarting += SetNonClientRegionSupport;
+}
+
+private void SetNonClientRegionSupport(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs args)
+{
+    var coreWebView2Settings = webView.CoreWebView2.Settings;
+    var urlCompareExample = "www.microsoft.com";
+    var uri = new Uri(args.Uri);
+
+    if (String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) &&
+        !coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = true;
+    }
+    else if (!String.Equals(uri.Host, urlCompareExample, StringComparison.OrdinalIgnoreCase) && 
+        coreWebView2Settings.IsNonClientRegionSupportEnabled)
+    {
+        coreWebView2Settings.IsNonClientRegionSupportEnabled = false;
+    }
+}
+```
+
+## Declaring Non-client App Regions
+Non-client regions are HTML elements that are marked with the css style `app-region`.
+* Draggable regions can be declared through the values `drag` or `no-drag`. 
+    * `app-region: drag` will support [draggable region functionality](#description) for the html element.
+    * `app-region: no-drag` will change cursor to I-bar, with text highlighting enabled. 
+    Elements with this style will not support draggable region functionality.
+```html
+<!DOCTYPE html>
+<body>
+    <div style="app-region:drag">Drag Region</div>
+    <div style="app-region:no-drag">No-drag Region</div>
```

Comment:
Better example would be to put the `no-drag` div *inside* the `drag` div, to show how you can counteract an outer `drag` with an inner `no-drag`. As written, the `no-drag` attribute is redundant since the initial value is already `no-drag`.

Reply:
Please update sample.

Reply:
updated